treeherder/client/thclient/client.py

```diff
@@ -709,11 +712,21 @@ def _get_uri(self, project, endpoint, data=None, oauth_key=None,
 
         return uri
 
+    def _get_uri(self, endpoint):
+        uri = '{0}://{1}/api/{2}'.format(
+            self.protocol, self.host, endpoint)
+
+        return uri
+
     def _get_json(self, project, endpoint, timeout, **params):
         if timeout is None:
             timeout = self.timeout
 
-        uri = self._get_uri(project, endpoint)
+        if project is None:
+            uri = self._get_uri(endpoint)
+        else:
+            uri = self._get_project_uri(project, endpoint)
+
```

Comment:
I have put an if condition here for checking project is None instead of making a separate method, since creating a separate function would just duplicate the code below. I think we can request _get_json function with None when there is no project... let me know if you think I should make a separate function for this.